tsinfer/inference.py

```diff
@@ -317,47 +332,81 @@ def match_ancestors(
     sample_data,
     ancestor_data,
     *,
-    num_threads=0,
-    path_compression=True,
     recombination_rate=None,
-    mismatch_rate=None,
-    precision=None,
-    extended_checks=False,
-    engine=constants.C_ENGINE,
-    progress_monitor=None,
+    mismatch_ratio=None,
+    recombination=None,  # Deliberately undocumented: allows passing prob array directly
+    mismatch=None,  # Deliberately undocumented: allows passing prob array directly
+    path_compression=True,
+    # performance-related parameters below
```

Comment:
I think it'd be make more sense to group the undocumented parameters together. Since they're all kwargs, the order isn't important for forward compatibility.  As above,  I don't think the classification of everything below as performance helps - progress_monitor and progress_monitor and engine and definitely not knobs to tweak to influence performance.

Reply:
OK. Happy to put whatever order you think is logical. As you say, since they are kwargs it's purely semantic.